### ISSUE TITLE
Salt-master too sensitive to whitespace in public keys (again)

### ISSUE THREAD

**Issue by justyns** [saltstack/salt#27750]
```
2015-10-06 16:53:07,968 [salt.utils.event                         ][DEBUG   ][23552] Sending event - data = {'_stamp': '2015-10-06T21:53:07.967800', 'result': False, 'pub': '-----BEGIN PUBLIC KEY-----\n<snip>\n-----END PUBLIC KEY-----', 'id': 'vm-123-123.testdomain.com'}
2015-10-06 16:53:07,967 [salt.transport.mixins.auth               ][ERROR   ][23552] Authentication attempt from vm-123-123.testdomain.com failed, the public keys did not match. This may be an attempt to compromise the Salt cluster.
2015-10-06 16:53:07,967 [salt.transport.mixins.auth               ][INFO    ][23552] Authentication request from vm-123-123.testdomain.com
```

I opened https://github.com/saltstack/salt/issues/21910 a while back, and it looks like the fixes merged in that issue have been reverted(or just not carried over to a new file.)   The same error cropped back up, but this time I had to edit `transport/mixins/auth.py` and add `.strip()` in a few different spots where the public key was read.

In the case of the error above, the key on the minion did not have a newline at the end of the file, but the key on the master did so they did not match despite the actual key being correct.

The salt-master version info:
```
[root@phx1-ca-saltmaster01 ~]# salt-call --versions
Salt Version:
           Salt: 2015.8.0

Dependency Versions:
         Jinja2: 2.2.1
       M2Crypto: 0.20.2
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
         Python: 2.6.6 (r266:84292, Jul 23 2015, 15:22:56)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed

System Versions:
           dist: centos 6.7 Final
        machine: x86_64
        release: 2.6.32-573.3.1.el6.x86_64
         system: CentOS 6.7 Final
```

https://github.com/saltstack/salt/issues/21910 has more details, but I wasn't sure how to re-open a GH issue ;o.   Assuming there wasn't a specific reason for not using `.strip()`, would a patch similar to the one in https://github.com/saltstack/salt/issues/21910 be accepted?

**Comment by jfindlay** [comment]
@justyns, thanks for reporting.  Your patch sounds reasonable and you are welcome to send it in (to the 2015.5 branch).  Also remember to reference this issue from the pull request.

**Comment by cachedout** [comment]
This should be fixed in #27979.
